packages/compass-preferences-model/src/provider.ts

```diff
@@ -1,18 +1,68 @@
 import { createContext, useContext } from 'react';
-import type { PreferencesAccess } from './';
-import { defaultPreferencesInstance } from '.';
+import type {
+  AllPreferences,
+  PreferenceStateInformation,
+  PreferencesAccess,
+  UserConfigurablePreferences,
+} from './';
+import { getDefaultsForAllPreferences } from './preferences-schema';
 export { usePreference, withPreferences } from './react';
 export { capMaxTimeMSAtPreferenceLimit } from './maxtimems';
 
+export class ReadOnlyPreferenceService implements PreferencesAccess {
+  private allPreferences: AllPreferences;
+  constructor(preferencesOverrides?: Partial<AllPreferences>) {
+    this.allPreferences = {
+      ...getDefaultsForAllPreferences(),
+      ...preferencesOverrides,
+    };
+  }
+
+  savePreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  refreshPreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  getPreferences() {
+    return this.allPreferences;
```

Comment:
This worked for this TODOed version of this implemementation, but I don't think we can just do that here. Preferences have dependencies on other preferences, and while curently this logic is part of the preferences-main implementation (see the `_computePreferenceValuesAndStates` method), it should be shared with this ReadOnlyPreferencesService implementation. This should also affect the `getPreferenceStates` and `getConfigurableUserPreferences` implementations

Reply:
Right, this is what was failing the existing test cases for me. Will push the change in a bit. The change accounts for the existence of deriveValue function on preference descriptors and use that to apply the derived values on top of defaults.